pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,470 @@
+package vz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	"github.com/Code-Hex/vz/v3"
+	"github.com/docker/go-units"
+	"github.com/lima-vm/lima/pkg/driver"
+	"github.com/lima-vm/lima/pkg/iso9660util"
+	"github.com/lima-vm/lima/pkg/limayaml"
+	"github.com/lima-vm/lima/pkg/localpathutil"
+	"github.com/lima-vm/lima/pkg/networks"
+	"github.com/lima-vm/lima/pkg/store/filenames"
+	"github.com/sirupsen/logrus"
+)
+
+func startVM(ctx context.Context, driver *driver.BaseDriver) (*vz.VirtualMachine, chan error, error) {
+	server, client, err := createSockPair()
+	if err != nil {
+		return nil, nil, err
+	}
+	machine, err := createVM(driver, client)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	fileConn, err := net.FileConn(server)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = machine.Start()
+
+	networks.StartGVisorNetstack(ctx, &networks.GVisorNetstackOpts{
+		Conn:         fileConn,
+		MTU:          1500,
+		SSHLocalPort: driver.SSHLocalPort,
+		MacAddress:   limayaml.MACAddress(driver.Instance.Dir),
+		Stream:       false,
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	errCh := make(chan error)
+	go func() {
+		//Handle errors via errCh and handle stop vm during context close
+
+		for {
+			select {
+			case <-ctx.Done():
+				logrus.Println("Context closed, stopping vm")
+				if machine.CanStop() {
+					_, err := machine.RequestStop()
+					logrus.Errorf("Error while stopping the VM %q", err)
+				}
+			case newState := <-machine.StateChangedNotify():
+				switch newState {
+				case vz.VirtualMachineStateRunning:
+					pidFile := filepath.Join(driver.Instance.Dir, filenames.QemuPID)
+					if _, err := os.Stat(pidFile); !errors.Is(err, os.ErrNotExist) {
+						logrus.Errorf("pidfile %q already exists", pidFile)
+						errCh <- err
+					}
+					if err := os.WriteFile(pidFile, []byte(strconv.Itoa(os.Getpid())+"\n"), 0644); err != nil {
+						logrus.Errorf("error writing to pid fil %q", pidFile)
+						errCh <- err
+					}
+					defer os.RemoveAll(pidFile)
+					logrus.Println("[VZ] - vm state change: running")
```

Comment:
Why are these `logrus.Println` vs `logrus.Info` or something else?

Reply:
Done the updates